pipeline_steps/run_tests.groovy

```diff
@@ -0,0 +1,39 @@
+def run_irr_tests() {
+  common.conditionalStage(
+    stage_name: "Run IRR Tests",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          ansiColor('xterm') {
+            dir(env.WORKSPACE) {
+              print("Triggered by PR: ${env.ghprbPullLink}")
+              checkout([$class: 'GitSCM',
+                branches: [[name: "origin/pr/${env.ghprbPullId}/merge"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'CleanCheckout']],
+                submoduleCfg: [],
+                userRemoteConfigs: [
+                  [
+                    url: "https://github.com/${env.ghprbGhRepository}.git",
+                    refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
+                  ]
+                ]
+              ]) // checkout
+              sh """#!/bin/bash
```

Comment:
Personally I don't think it'll make that much difference as it all stands now. It may be an idea later to change the rpccheckout method to be more generic and then re-use it, but for now I think that's probably overkill.